src/modules/simulation/simulator_mavlink/SimulatorMavlink.cpp

```diff
@@ -1432,6 +1435,24 @@ void SimulatorMavlink::check_failure_injections()
 				supported = true;
 				_airspeed_blocked = false;
 			}
+
+		} else if (failure_unit == vehicle_command_s::FAILURE_UNIT_SENSOR_VIO) {
+			handled = true;
+
+			if (failure_type == vehicle_command_s::FAILURE_TYPE_OFF) {
+				PX4_WARN("CMD_INJECT_FAILURE, vio off");
+				supported = true;
+				_vio_blocked = true;
+
+			} else if (failure_type == vehicle_command_s::FAILURE_TYPE_OK) {
+				PX4_INFO("CMD_INJECT_FAILURE, vio ok");
+				supported = true;
+				_vio_blocked = false;
+			}
+		}
+
+		if (!supported) {
+			PX4_WARN("CMD_INJECT_FAILURE, failure sensor/type combination not supported");
```

Comment:
The problem with this is that there might be more places where the failure commands are handled, and not just here. So it might lead to the confusing case where you see this error but then the failure still "works".

Reply:
Ah, I see.  Yeah, that could be confusing.